ui/src/features/common/promotion-directive-step-status/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { PromotionStatusPhase } from '@ui/features/common/promotion-status/utils';
+import { PromotionStatus } from '@ui/gen/v1alpha1/generated_pb';
+
+// UI concludes from Promotion's status data
+export enum PromotionDirectiveStepStatus {
+  RUNNING,
+  FAILED,
+  SUCCESS
+}
+
+export const getPromotionDirectiveStepStatus = (
+  stepNumber: number,
+  promotionStatus?: PromotionStatus
+) => {
+  if (
+    promotionStatus?.phase === PromotionStatusPhase.RUNNING &&
+    stepNumber === Number(promotionStatus?.currentStep)
+  ) {
+    return PromotionDirectiveStepStatus.RUNNING;
+  }
+
+  // TODO: controller should point 'currentStep' to exact failed step
+  // at the moment, it is defaulted to 0
```

Comment:
I can fix that right now.

Reply:
Actually, I could only find this being set to zero if the steps couldn't even get started or after success. I updated it so it's _not_ set back to zero on success.

I cannot seem to find where a failed step is getting this set back to zero.